### PR TITLE
[WIP] add source link support to portable pdbs

### PR DESCRIPTION
**Pull request by ctaggart**
I thought EntityFramework would make a good example for an E2E experience with Portable PDBs that have source link support. https://github.com/dotnet/roslyn/issues/12759

To get an idea what these changes do, please read the readme that I updated today:
https://github.com/ctaggart/SourceLink

Then try:
```
git clone git@github.com:ctaggart/EntityFramework.git -c core.autocrlf=input
cd .\EntityFramework\src\Microsoft.EntityFrameworkCore.Relational.Design
dotnet build
dotnet sourcelink test .\bin\Debug\netstandard1.3\Microsoft.EntityFrameworkCore.Relational.Design.pdb
```

Example:
```
PS C:\Users\camer\cs\EntityFramework\src\Microsoft.EntityFrameworkCore.Relational.Design> dotnet sourcelink test .\bin\Debug\netstandard1.3\Microsoft.EntityFrameworkCore.Relational.Design.pdb
sourcelink test passed
PS C:\Users\camer\cs\EntityFramework\src\Microsoft.EntityFrameworkCore.Relational.Design> dotnet sourcelink test .\bin\Debug\netstandard1.3\Microsoft.EntityFrameworkCore.Relational.Design.pdb
sourcelink test passed
PS C:\Users\camer\cs\EntityFramework\src\Microsoft.EntityFrameworkCore.Relational.Design> dotnet sourcelink print-json .\bin\Debug\netstandard1.3\Microsoft.EntityFrameworkCore.Relational.Design.pdb
{"documents":{"C:\\Users\\camer\\cs\\EntityFramework\\*":"https://raw.githubusercontent.com/ctaggart/EntityFramework/f36ce80a93d9f698ff6b70207d2b83b5b11cf93b/*"}}
PS C:\Users\camer\cs\EntityFramework\src\Microsoft.EntityFrameworkCore.Relational.Design> dotnet sourcelink print-urls .\bin\Debug\netstandard1.3\Microsoft.EntityFrameworkCore.Relational.Design.pdb
ab1db8972c75cdcc0680a87e3c7e5048fa493a5c sha1 csharp C:\Users\camer\cs\EntityFramework\src\Shared\SharedTypeExtensions.cs
https://raw.githubusercontent.com/ctaggart/EntityFramework/f36ce80a93d9f698ff6b70207d2b83b5b11cf93b/src/Shared/SharedTypeExtensions.cs
...
```

Including the Portable PDBs in the package is doable. It would be easier if `--include-portable-pdbs` was added. https://github.com/NuGet/Home/issues/4142#issuecomment-282573801

### CODE REVIEW

File: tools/SourceLink.props [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<Project>
+  <PropertyGroup>
+    <SourceLinkCreate>true</SourceLinkCreate>
```
Comment (ctaggart) [inline tools/SourceLink.props]: Forcing it on with <SourceLinkCreate>true</SourceLinkCreate> is just to show that it works. The line should be removed before a merge.

**Comment by bricelam** [comment]
@Eilon Has there been any discussion of doing this all-up on the ASP.NET packages?

**Comment by Eilon** [comment]
We should discuss. Whatever we do, we should do consistently in all the repos.

**Comment by ctaggart** [comment]
@bricelam @Eilon  @DamianEdwards the plan has been discussed in this issue: https://github.com/aspnet/Universe/issues/131#issuecomment-154227813

**Comment by ctaggart** [comment]
Closing this pull request. It was just an example of how easy it is to add.

**Comment by paulomorgado** [comment]
What's preventing this from happening?

**Comment by Eilon** [comment]
@paulomorgado can you provide info on what isn't working? I think the PDBs are source linked and on the Microsoft symbol server so in theory everything should be working.

**Comment by paulomorgado** [comment]
@Eilon, I tried to set up a function breakpoint inside EF Core and tried to step into `ToListAsync` and nothing happened.

Besides the fact that everyone these days are using embedded PDBs or packaging them with the libraries and linking the PBDs to the source (GitHub, in this case), what do I have to do to step into EF Core's code?

**Comment by Eilon** [comment]
@paulomorgado I saw in another issue that there was an issue with the symbol servers just recently and that it started working again.

I found a blog post describing the settings you need to enable the source stepping to work: https://www.stevejgordon.co.uk/debugging-asp-net-core-2-source

Have you followed those steps and it's still not working?

**Comment by paulomorgado** [comment]
I had the **Microsoft Symbol Servers** location unchecked because

> The performance of remote symbol stores can vary due to latency, bandwidth, or availability of servers.  Debugging performance can be improved by selecting the option to load only specified modules when enabling this server.

You really should make the change to https://github.com/dotnet/sourcelink

**Comment by Eilon** [comment]
@paulomorgado sorry I'm not sure what change you are proposing, can you provide more details?

**Comment by Eilon** [comment]
@paulomorgado there was a discussion a long time ago and we concluded that the overall best solution in terms of supportability and functionality was to use symbol servers. It works cross platform, and it works in the broadest set of tools, and doesn't affect anyone not interested in symbols.

**Pull request closed without merging**